CMakeLists.txt

```diff
@@ -286,4 +293,59 @@ if(ENABLE_DEPS)
+			BUILD_COMMAND 		"${CMAKE_COMMAND}" "--build" "<BINARY_DIR>" --config "Release"
+			INSTALL_COMMAND 	"${CMAKE_COMMAND}" "--install" "<BINARY_DIR>" --config "Release"
+		)
+		ExternalProject_Get_property(vulkan_headers INSTALL_DIR)
+		set(vulkan_headers_install_dir "${INSTALL_DIR}")
+
+		ExternalProject_Add(vulkan_loader
+			DEPENDS 			spirv_headers vulkan_headers
+			GIT_REPOSITORY 		"https://github.com/KhronosGroup/Vulkan-Loader.git"
+			GIT_TAG 			"main"
+			CMAKE_ARGS 			"-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_BINDIR=bin" "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>" "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}" "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}" "-DCMAKE_PREFIX_PATH=${vulkan_headers_install_dir}"
+			BUILD_COMMAND 		"${CMAKE_COMMAND}" "--build" "<BINARY_DIR>" --config "Release"
+			INSTALL_COMMAND 	"${CMAKE_COMMAND}" "--install" "<BINARY_DIR>" --config "Release"
+		)
+		ExternalProject_Get_property(vulkan_loader INSTALL_DIR)
+		set(vulkan_loader_install_dir "${INSTALL_DIR}")
+
+		find_package(BISON REQUIRED)
+		ExternalProject_Add(vkd3d
+			DEPENDS 			spirv_headers vulkan_headers vulkan_loader
+			URL					"https://dl.winehq.org/vkd3d/source/vkd3d-1.13.tar.xz"
+			URL_HASH			"SHA256=cf637873695fcc02fab308f68608f0bbb90481332a2ff0a9597c7c9fd97d363d"
+			DOWNLOAD_EXTRACT_TIMESTAMP "1"
+			CONFIGURE_COMMAND 	"sh" "<SOURCE_DIR>/configure" "--prefix=<INSTALL_DIR>" "--enable-tests=no" "--enable-demos=no" "--disable-doxygen-doc" "CFLAGS=-I${spirv_headers_install_dir}/include -I${vulkan_headers_install_dir}/include -I${vulkan_loader_install_dir}/include" "LDFLAGS=-L${vulkan_loader_install_dir}/lib" "BISON=${BISON_EXECUTABLE}"
+			BUILD_COMMAND	 	"make"
+			INSTALL_COMMAND	 	"make" "install"
+		)
+		ExternalProject_Get_property(vkd3d SOURCE_DIR)
+		ExternalProject_Get_property(vkd3d INSTALL_DIR)
+		install(DIRECTORY "${INSTALL_DIR}/lib/" DESTINATION "${CMAKE_INSTALL_LIBDIR}" FILES_MATCHING PATTERN "*.so*" PERMISSIONS ${chmod_0755})
+		install(DIRECTORY "${INSTALL_DIR}/lib/" DESTINATION "${CMAKE_INSTALL_LIBDIR}" FILES_MATCHING PATTERN "*.dylib*" PERMISSIONS ${chmod_0755})
+		install(FILES "${SOURCE_DIR}/COPYING" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/licenses/vkd3d")
+	endif()
+	if(LINUX)
+		find_program(PATCHELF_BIN NAMES "patchelf" REQUIRED)
+		install(CODE "file(GLOB so_paths \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/*so*\")\n  foreach(so_path \${so_paths})\n    if(NOT IS_SYMLINK \${so_path})\n      message(STATUS \"Adding \\\"\$ORIGIN\\\" to RPATH of \${so_path}\")\n      execute_process(COMMAND ${PATCHELF_BIN} \"\${so_path}\" --add-rpath \"\$ORIGIN\")\n    endif()\n  endforeach()")
+	elseif(APPLE)
+		# FIXME: Apple probably needs to do something similar as Linux, but using otool
```

Comment:
This is ready.

Using shadercross of a linux archive, I could create dxbc shaders for the SDL_ttf example https://github.com/libsdl-org/SDL_ttf/pull/412

There might still be an issue with macOS due to missing rpaths.